xtrack/line.py

```diff
@@ -1113,20 +1318,30 @@ def get_elements_of_type(self, types):
 
         return elements, names
 
-    def check_aperture(self):
+    def check_aperture(self, needs_aperture=[]):
 
         '''Check that all active elements have an associated aperture.'''
 
         elements_df = self.to_pandas()
 
-        elements_df['is_aperture'] = elements_df.element_type.map(lambda s: s.startswith('Limit'))
+        elements_df['is_aperture'] = elements_df.name.map(
+                                            lambda nn: _is_aperture(self.element_dict[nn]))
         elements_df['i_aperture_upstream'] = np.nan
         elements_df['s_aperture_upstream'] = np.nan
         elements_df['i_aperture_downstream'] = np.nan
         elements_df['s_aperture_downstream'] = np.nan
-
         num_elements = len(self.element_names)
 
+        # Elements that don't need aperture
+        dont_need_aperture = {name: False for name in elements_df['name']}
+        for name in elements_df['name']:
+            ee = self.element_dict[name]
+            if _allow_backtrack(ee) and not name in needs_aperture:
+                dont_need_aperture[name] = True
+            # Correct isthick for elements that need aperture but have zero length:
+            if name in needs_aperture and hasattr(ee, 'length') and ee.length == 0:
+                elements_df.loc[elements_df['name']==name, 'isthick'] = False
```

Comment:
This is unclear to me. Maybe mention an example in the comments explaining why this is needed

Reply:
Added a comment